mcs/class/Mono.Data.Tds/Test/bug-4786.cs

```diff
@@ -51,13 +51,10 @@ public void CheckNullException()
 	//require at this point: a listener on port 1433...
 
 	try{
-		Socket Listener = new Socket(AddressFamily.InterNetwork, 
-                                         SocketType.Stream,
-                                         ProtocolType.Tcp);
 		IPAddress hostIP =Dns.GetHostEntry("localhost").AddressList[0];
-        	IPEndPoint ep = new IPEndPoint(hostIP, 1433);
-		Listener.Bind(ep); 
-        	Listener.Listen(1);
+        IPEndPoint ep = new IPEndPoint(hostIP, 1433);
+        TcpListener Listener = new TcpListener (ep);
+        Listener.Start ();
```

Comment:
The `Listener` is never stopped.

Reply:
It will be closed at Dispose. But It might be a good idea to use a "using" block, that will ensure we do the correct thing in any possible escenario.

Reply:
Pushing with a stop, will make the code more readable and the change will be smaller.